src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -718,3 +726,27 @@
              {:current-screen stack-id
               :start-flow?    start-flow?
               :flow-id        flow-id}]]]})))
+
+(rf/reg-event-fx
+ :wallet/transaction-confirmation-navigate-back
+ (fn [{db :db} [{:keys []}]]
+   (let [tx-type       (-> db :wallet :ui :send :tx-type)
+         keep-tx-data? (#{:account-collectible-tab :wallet-stack}
+                        (-> db :wallet :ui :send :entry-point))]
+     {:db (cond-> db
+            (and (= tx-type :tx/collectible-erc-721) (not keep-tx-data?))
+            (update-in [:wallet :ui :send] dissoc :tx-type :amount :route :suggested-routes)
+
+            (= tx-type :tx/collectible-erc-1155)
+            (update-in [:wallet :ui :send] dissoc :route :suggested-routes))
+      :fx [[:dispatch [:navigate-back]]]})))
+
+(rf/reg-event-fx
+ :wallet/collectible-amount-navigate-back
+ (fn [{db :db} [{:keys []}]]
+   (let [keep-tx-data? (#{:account-collectible-tab :wallet-stack}
+                        (-> db :wallet :ui :send :entry-point))]
+     {:db (cond-> db
+            :always             (update-in [:wallet :ui :send] dissoc :amount :route)
+            (not keep-tx-data?) (update-in [:wallet :ui :send] dissoc :tx-type))
+      :fx [[:dispatch [:navigate-back]]]})))
```

Comment:
These new events remove info from DB when we navigate back, so that the account switcher can be hidden/shown properly.

This is a great example of a behavior that should be handled by a new infrastructure for the wizard